[ios] Move keyboard and keyboardview to ios-common

fuzzard opened the pull request:
common keyboard related code moved to ios-common
remove XBMCDebugHelpers.h and use common NSLogDebugHelpers.h in platform/darwin

## Description
move/update IOSKeyboard*.* from platform/darwin/ios to platform/darwin/ios-common
remove XBMCDebugHerlpers.h and use the same file that exists in platform/darwin/NSLogDebugHelpers.h

## Motivation and Context
code is very similar to what will be used in the ATV PR, so i think itd be best to move it to the common folder in advance of the ATV PR. Doesnt have any functional changes with existing code, just relocates and updates header includes

Have avoided any style changes as best i can to keep the PR as simple as possible for now.

## Types of change
<!--- What type of change does your code introduce? Put an `x` in all the boxes that apply like this: [X] -->
- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [X] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

fuzzard committed:
[ios] Move keyboard and keyboardview to ios-common

common keyboard related code moved to ios-common

fuzzard commented:
Is it worth running a clang-format over the relocated files (second commit obviously) to get them upto current kodi code guidelines?

Rechi commented on xbmc/platform/darwin/ios-common/IOSKeyboard.mm:
```diff
@@ -6,11 +6,12 @@
+#include "platform/darwin/ios/XBMCController.h"
+#include "platform/darwin/ios-common/IOSKeyboard.h"
+#include "platform/darwin/ios-common/IOSKeyboardView.h"
+
```
no need for another empty line

Rechi commented on xbmc/platform/darwin/ios/XBMCDebugHelpers.h:
```diff
@@ -1,28 +0,0 @@
-/*
```
move this into a separate commit

fuzzard replied:
Done

fuzzard committed:
[ios] remove XBMCDebughelpers.h

existing duplicate file located in platform/darwin/NSLogDebugHelpers.h
use it instead of carrying duplicate code

fuzzard commented:
Thanks for the look Rechi. Have created the second commit for the debughelper change
